### ISSUE TITLE
Navigation Block - Add setting to show submenus on arrow button click (not hover)

### ISSUE THREAD

**Issue by patrickwc** [WordPress/gutenberg#65214]
## What problem does this address?
At the moment it is not possible to add a navigation block with submenus that show on click if you _also_ want that link to be a normal a href link. I know you can toggle Submenus -> Open on Click in the navigation block settings, but this changes the menu item link into a button, meaning you can no longer access that menu item link.

It should be possible to have a navigation link with a submenu arrow, where the link text is clickable as a link and the arrow on the right acts as a button to show/hide the submenus.

## What is your proposed solution?
Add a setting which allows a user to disable showing a submenu on hover, only showing the menu on submenu-icon (arrow) click. If this was selected, the user would have to click the arrow to show the submenu.

In WP 6.3 it was possible to override this behaviour with CSS, but since WP 6.4 JavaScript is used to change the `.wp-block-navigation-submenu__toggle`'s aria-expanded attribute to true, so you cannot override this behaviour (to my knowledge).

I have done this myself by overwriting the CSS in WP 6.3, but because the JS listening event triggers a change to the `aria-expanded` value on _hover_ (since WP 6.4, I don't know what benefit this has...), I had to overwrite all of the CSS which triggers the menu to show on `aria-expanded=true` and create my own JS listener event with a custom data-expanded attribute and my own custom CSS (as well as overriding the core CSS) to get submenus to show on submenu toggle click.

**Comment by talldan** [comment]
This seems like a valid request, but I expect an issue is that the arrow currently in use is not big enough to be accessible in most themes, it's only about 10px x 10px in Twenty Twenty Four.

It might be ok to switch to chunkier arrows when this particular setting is enabled.

As an aside, I also noticed when 'Open on click' is set on the navigation block, the link button is hidden from the toolbar of submenu blocks to indicate links don't work, but it can still be set in the sidebar 🤦

**Comment by patrickwc** [comment]
Great. You can always add padding to the button arrow to make the button accessible, making the button ~44px x 44px with the svg arrow still at 10px x 10px in the middle if needed.

To be it doesn't make much sense that a link defined in wp-admin -> Appearance -> Menus is then disabled in the navigation block with "Open on click". It was a link, why would a user not want it to be a link now?

**Comment by ohfuchs** [comment]
I came here to request this as well, along with an option to always expand by default (nested list style). This would also allow for easier customization.

**Comment by joedolson** [comment]
It is an accessibility issue that the arrow by itself is not large enough as a click target, but I agree that it would be good to have the option to use the top level link as a link and have a button to expand the submenu - this is a pretty common pattern, and should be supported. 

Honestly, the styles controlling that dropdown arrow in the navigation block styles are fairly strange; there's a lot of CSS there specifically to force the button to be smaller. Much the same visual effect can be accomplished with a larger button by removing `height: .6em`, `line-height: 0`, `padding: 0`, and `width: .6em` from `.wp-block-navigation .wp-block-navigation__submenu-icon`.

**Comment by DecimusBurton** [comment]
Here is what I think you are looking for [in action live on our hobby site](https://www.bajanthings.com/wordpress-6-7-first-impressions/#webd).  Photo of desktop and mobile navigation menus in action.
